geoportal/c2cgeoportal_geoportal/lib/xsd.py

```diff
@@ -81,7 +81,15 @@ def add_association_proxy_xsd(self, tb, column_property):
         association_proxy = getattr(cls, proxy)
         relationship_property = class_mapper(cls).get_property(association_proxy.target)
         target_cls = relationship_property.argument
-        query = DBSession.query(getattr(target_cls, association_proxy.value_attr))
+        if (
+            hasattr(cls, "__enumerations_order__")
+            and association_proxy.value_attr in cls.__enumerations_order__
+        ):
+            query = DBSession.query(getattr(target_cls, association_proxy.value_attr)).order_by(
+                getattr(target_cls, cls.__enumerations_order__[association_proxy.value_attr])
+            )
+        else:
+            query = DBSession.query(getattr(target_cls, association_proxy.value_attr))
```

Comment:
Note that association_proxy.value_attr is not the name of the enum in main table, but name of the column to use as a value in foreign table.
Currently, in association proxy you can only access target (relation) and value_attr (name column).
But the name of the relation is not trivial to find for admins, it depends on multiple things.
And the interesting column name is only stored in the left side of the relation.primaryjoin condition.
If it cannot be easily accessed, I would recommend to add a property to the association proxy with the main table foreign key column name at construction time.

Reply:
Personnaly I would vote for putting directly the order attribute in AssociationProxy ex:
```
    sort_column = None
    if (
        hasattr(cls, "__enumerations_order__")
        and col.name in cls.__enumerations_order__
        and 
    ):
        sort_column = getattr(child_cls, cls.__enumerations_order__[col.name]
    setattr(cls, proxy, _AssociationProxy(rel, "name", nullable=nullable, sort=sort_column))
```

Reply:
Between both formula, this one looks clearer yes.

But I don't understand why we would need this line.
I've updated my code. It looks working without that.

I've tried to add the `order_by` on the `query` only in the `if` statement (your first comment) but I doesn't work (and I know that works when we do things like `query.all()`). Perhaps that doesn't work here because we use the query as a iterator ?

Reply:
(Tests are also better now)